Api/Api.cs

```diff
@@ -1184,6 +1187,226 @@ public Organization ReadOrganization(string organizationId = null)
+            string strategy,
+            string compileId,
+            HashSet<OptimizationParameter> parameters,
+            IReadOnlyList<Constraint> constraints)
+        {
+            var request = new RestRequest("optimizations/estimate", Method.POST)
+            {
+                RequestFormat = DataFormat.Json
+            };
+
+            request.AddParameter("application/json", JsonConvert.SerializeObject(new
+            {
+                projectId,
+                name,
+                target,
+                targetTo,
+                targetValue,
+                strategy,
+                compileId,
+                parameters,
+                constraints
+            }), ParameterType.RequestBody);
+
+            ApiConnection.TryRequest(request, out EstimateResponseWrapper response);
+            return response.Estimate;
+        }
+
+        /// <summary>
+        /// Create an optimization with the specified parameters via QuantConnect.com API
+        /// </summary>
+        /// <param name="projectId">Project ID of the project the optimization belongs to</param>
+        /// <param name="name">Name of the optimization</param>
+        /// <param name="target">Target of the optimization</param>
+        /// <param name="targetTo">Target extremum of the optimization</param>
+        /// <param name="targetValue">Optimization target value</param>
+        /// <param name="strategy">Optimization strategy</param>
+        /// <param name="compileId">Optimization compile ID</param>
+        /// <param name="parameters">Optimization parameters</param>
+        /// <param name="constraints">Optimization constraints</param>
+        /// <param name="estimatedCost">Estimated cost for optimization</param>
+        /// <param name="nodeType">Optimization node type</param>
+        /// <param name="parallelNodes">Number of parallel nodes for optimization</param>
+        /// <returns>BaseOptimization object from the API.</returns>
+        public BaseOptimization CreateOptimization(
+            int projectId,
+            string name,
+            string target,
+            string targetTo,
+            decimal? targetValue,
+            string strategy,
+            string compileId,
+            HashSet<OptimizationParameter> parameters,
+            IReadOnlyList<Constraint> constraints,
+            decimal estimatedCost,
+            string nodeType,
+            int parallelNodes)
+        {
+            var request = new RestRequest("optimizations/create", Method.POST)
+            {
+                RequestFormat = DataFormat.Json
+            };
+
+            request.AddParameter("application/json", JsonConvert.SerializeObject(new
+            {
+                projectId,
+                name,
+                target,
+                targetTo,
+                targetValue,
+                strategy,
+                compileId,
+                parameters,
+                constraints,
```

Comment:
Think something of with the constraints format?
![image](https://user-images.githubusercontent.com/18473240/146095229-aca508d1-c1e3-4a68-afba-387e870e552b.png)
![image](https://user-images.githubusercontent.com/18473240/146095242-cd1d4b90-77e3-4ce7-a1cc-fd30a40805d5.png)

Reply:
Changing how we deserialize constraints is not really part of this task, but I'm more than happy to change it.

Reply:
Resolved server-side.